File: cocos2d-html5_i/IronCity/src/layer/GameSceneMenuLayer.js
/**
 * Created with JetBrains WebStorm.
 * User: cocos
 * Date: 13-10-23
 * Time: 下午1:56
 * To change this template use File | Settings | File Templates.
 */

//layer: game menu.
var GameSceneMenuLayer = cc.UILayer.extend({
    parentScene:null,
    settingBtn:null,
    bloodBar:null,
    distanceScore:null,
    musicEffect:null,
    musicVolume:null,
    init:function(bloodBarPercent, value){
        if( cc.UILayer.prototype.init.call(this) ){
            this.parentScene = GameScene.getScene();
            this.addWidget( cc.UIHelper.getInstance().createWidgetFromJsonFile(Json_IronCityUI_1) );

            this.settingBtn    = this.getWidgetByName("Setting");
            this.bloodBar      = this.getWidgetByName("BloodBar");
            this.distanceScore = this.getWidgetByName("DistanceScore");

            this.settingBtn.addTouchEventListener(this, this.settingBtnCallback);
            this.setBloodBarPercent(bloodBarPercent);
            this.setDistanceScore(value);

            this.musicEffect = 0;
            this.musicVolume = 50;

            return true;
        }
        return false;
    },
    //set blood.
    setBloodBarPercent:function(percent){
        this.bloodBar.setPercent(percent);
    },
    //set distance score.
    setDistanceScore:function(value){
        this.distanceScore.setStringValue(value);
    },
    //call back function of setting button.
    settingBtnCallback:function(pSender, type){
        //if(cc.TouchEventType.BEGAN == type){
        if(cc.TouchEventType.ENDED == type){
            this.parentScene = GameScene.getScene();
            var gameSetLayer = new GameSceneSetLayer();
            gameSetLayer.init(this.musicEffect, this.musicVolume);

            gameSetLayer.setAnchorPoint(cc.p(0, 0));
            gameSetLayer.setPosition(cc.p(0, 0));

            //console.log("scene:", this.parentScene, "setLayer: ", gameSetLayer);
            this.parentScene.addChild(gameSetLayer,4);
            this.parentScene.pause();
            //console.log("click set, and goto setLayer.");
            return true;
        }
    },
    //get score of distance.
    getDistanceScore:function(){
        return this.distanceScore.getStringValue();
    }
});

File: cocos2d-x-2.2.0/projects/Ironman/Classes/GameSceneMenuLayer.cpp
//
//  GameSceneMenuLayer.cpp
//  Ironman
//
//  Created by Ken on 13-10-14.
//
//

#include "GameSceneMenuLayer.h"

bool GameSceneMenuLayer::init(int broodBarPercent,const char *value)
{
    if(UILayer::init()){
        
        parentScene = GameScene::shareGameScene();
        this->addWidget(dynamic_cast<Layout*>(CCUIHELPER->createWidgetFromJsonFile("iphone/IronCityUI_1.json")));
        
        settingBtn    = dynamic_cast<UIButton*>(this->getWidgetByName("Setting"));
        broodBar      = dynamic_cast<UILoadingBar*>(this->getWidgetByName("BroodBar"));
        distanceScore = dynamic_cast<UILabelAtlas*>(this->getWidgetByName("DistanceScore"));
		/*
		settingBtn->setAnchorPoint(ccp(0,1));
		broodBar->setAnchorPoint(ccp(0,1));
		distanceScore->setAnchorPoint(ccp(0,1));

		settingBtn->setPosition(ccp(
		::rightTop().x+settingBtn->getPosition().x,VisibleRect::rightTop().y-settingBtn->getPosition().y-150));
		broodBar->setPosition(ccp(VisibleRect::rightTop().x+broodBar->getPosition().x,VisibleRect::rightTop().y-broodBar->getPosition().y-150));
		distanceScore->setPosition(ccp(VisibleRect::rightTop().x+distanceScore->getPosition().x,VisibleRect::rightTop().y-distanceScore->getPosition().y-150));
		*/
        settingBtn->addTouchEventListener(this, toucheventselector(GameSceneMenuLayer::settingBtnCallback));
        this->setBroodBarPercent(broodBarPercent);
        this->setDistanceScore(value);
		
		musicEffect = 0;
		musicVolume = 50;
        
        return true;
    }
    return false;
}

void GameSceneMenuLayer::settingBtnCallback(CCObject *pSender, TouchEventType type)
{
	if(TOUCH_EVENT_BEGAN == type){
		
		GameSceneSetLayer* gameSetLayer = new GameSceneSetLayer();
		if(gameSetLayer && gameSetLayer->init(musicEffect, musicVolume)){
			
			gameSetLayer->autorelease();
		}else{
			
			CC_SAFE_DELETE(gameSetLayer);
		}
		
		gameSetLayer->setAnchorPoint(ccp(0, 0));
		gameSetLayer->setPosition(ccp(0,0));
		
		parentScene->addChild(gameSetLayer,4);
	}
	
}

void GameSceneMenuLayer::setBroodBarPercent(int percent)
{
    broodBar->setPercent(percent);
}

void GameSceneMenuLayer::setDistanceScore(const char *value)
{
    distanceScore->setStringValue(value);
}

const char* GameSceneMenuLayer::getDistanceScore()
{
    return distanceScore->getStringValue();
}